Source/Core/AbsyExpr.cs

```diff
@@ -169,7 +169,7 @@ public static NAryExpr Binary(BinaryOperator.Opcode op, Expr e0, Expr e1)
       Contract.Requires(e1 != null);
       Contract.Requires(e0 != null);
       Contract.Ensures(Contract.Result<NAryExpr>() != null);
-      return Binary(Token.NoToken, op, e0, e1);
+      return Binary(e0.tok ?? Token.NoToken, op, e0, e1);
```

Comment:
This looks fine and the `??` is a good catch. If one wanted to go one step further, I suppose one could write

``` cs
e0.tok ?? e1.tok ?? Token.NoToken
```